modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/impl/TestMvTableStorage.java

```diff
@@ -180,33 +151,51 @@ public void destroy() throws StorageException {
     }
 
     @Override
-    public CompletableFuture<Void> startRebalanceMvPartition(int partitionId) {
-        MvPartitionStorage oldPartitionStorage = partitions.get(partitionId);
+    public CompletableFuture<Void> startRebalanceMvPartition(int partitionId) throws StorageException {
+        Storages oldStorages = storagesByPartitionId.get(partitionId);
+
+        checkPartitionStoragesExists(oldStorages, partitionId);
 
-        assert oldPartitionStorage != null : "Partition does not exist: " + partitionId;
+        if (backupStoragesByPartitionId.putIfAbsent(partitionId, oldStorages) == null) {
+            Storages newStorages = new Storages(new TestMvPartitionStorage(partitionId));
 
-        if (backupPartitions.putIfAbsent(partitionId, oldPartitionStorage) == null) {
-            partitions.put(partitionId, new TestMvPartitionStorage(partitionId));
+            oldStorages.hashIndexes.keySet().forEach(indexId -> newStorages.hashIndexes.put(
```

Comment:
Creating indexes is not mandatory, as far as I see. Rebalance itself will create indexes that it needs

Reply:
I suggest discussing this in person.

Reply:
We decided personally that we would recreate them.